### ISSUE TITLE
Manually reload async belongsTo relationship

### ISSUE THREAD

**Issue by jcbvm** [emberjs/data#2380]
It would be nice to have an option to manually reload a belongsTo relationship the same way we can reload a hasMany relationship (`model.get('relationship').reload()`).

Currently we'll have to do it this way:
```js
model.get('relationship').then(function(relationship) {
    relationship.reload();
});
```
The problem with this is that we first have to load it before we can reload it which may cause loading it twice.

(Also reloading it this way does not work when the belongsTo has a link.)

**Comment by jcbvm** [comment]
I still opt for a reloadRelationship function. So instead of calling reload on the relationship itself, we should be able to call reload from it's parent which will take the context of the parent into account. This way, reloading can check for links.

**Comment by igorT** [comment]
We definitely want to support this, here is a RFC that tracks this https://github.com/emberjs/rfcs/pull/57

**Comment by jcbvm** [comment]
Thanks, will keep an eye on it!

**Comment by pangratz** [comment]
I think this issue can be closed since an [initial implementation of references](https://github.com/emberjs/data/pull/3303) has been merged into `master`. Having the `ds-references` feature enabled, a relationship can be reloaded via `model.belongsTo("family").reload()` and `model.hasMany("children").reload()`.

**Comment by fivetanley** [comment]
I agree with @pangratz. Keep an eye out for references work.
